tools/scenario-player/scenario_player/scenario.py

```diff
@@ -0,0 +1,313 @@
+import pathlib
+from collections.abc import Mapping
+from typing import Dict, List, Tuple, Any, Union
+
+import structlog
+import yaml
+from scenario_player.exceptions import (
+    ScenarioError,
+    MissingNodesConfiguration,
+    MultipleTaskDefinitions,
+    InvalidScenarioVersion,
+)
+from scenario_player.constants import (
+    DEFAULT_TOKEN_BALANCE_MIN,
+    DEFAULT_TOKEN_BALANCE_FUND,
+    OWN_ACCOUNT_BALANCE_MIN,
+    NODE_ACCOUNT_BALANCE_MIN,
+    NODE_ACCOUNT_BALANCE_FUND,
+    TIMEOUT,
+    API_URL_ADDRESS,
+    API_URL_TOKENS,
+    API_URL_TOKEN_NETWORK_ADDRESS,
+    SUPPORTED_SCENARIO_VERSIONS,
+)
+from scenario_player.tasks.base import get_task_class_for_type
+from scenario_player.utils import get_gas_prize_strategy
+log = structlog.get_logger(__name__)
+
+
+class NodesConfig(Mapping):
```

Comment:
Is there a need to inherit from `Mapping`? I only see attribute access being used.

Reply:
It originally was mapping-like, and I figured it'd be less surprising if it would still behave like one. This possibly holds more true for the `Scenario`class than this one, I suppose though. It might also be more convenient down the road for development if we can simply access the configuration by the keys set in the yaml, rather than having to specify each possible attribute in the `__init__`.  

This could, of course, be done via a `__getattr__` modification as well, but I believe this approach to be cleaner.